date.go

```diff
@@ -132,10 +132,18 @@ func (d *dateObject) unset() {
 }
 
 func (d *dateObject) time() time.Time {
+	ti, ok := checkTime(d.val.__wrapped)
+	if ok {
```

Comment:
[opt] you could combine these into a single line. Same goes for the `timeUTC` function below.